File: java/src/demos/dlineage/dataflow/model/xml/relation.java

package demos.dlineage.dataflow.model.xml;

import java.util.List;

import org.simpleframework.xml.Attribute;
import org.simpleframework.xml.Element;
import org.simpleframework.xml.ElementList;

public class relation implements Cloneable
{

	@Attribute(required = false)
	private String id;

	@Attribute(required = false)
	private String type;

	@Element(name = "target", required = false)
	private targetColumn target;

	@ElementList(entry = "source", inline = true, required = false)
	private List<sourceColumn> sources;

	public String getId( )
	{
		return id;
	}

	public void setId( String id )
	{
		this.id = id;
	}

	public String getType( )
	{
		return type;
	}

	public void setType( String type )
	{
		this.type = type;
	}

	public targetColumn getTarget( )
	{
		return target;
	}

	public void setTarget( targetColumn target )
	{
		this.target = target;
	}

	public List<sourceColumn> getSources( )
	{
		return sources;
	}

	public void setSources( List<sourceColumn> sources )
	{
		this.sources = sources;
	}

	public boolean isDataFlow( )
	{
		return "dataflow".equals( type );
	}

	@Override
	public Object clone( ) throws CloneNotSupportedException
	{
		return super.clone( );
	}
}

File: dotnet/src/demos/dataFlowAnalyzer/dataflow/model/xml/relation.cs
using System.Collections.Generic;

namespace gudusoft.gsqlparser.demos.dlineage.dataflow.model.xml
{
    using System.Xml.Serialization;

    public class relation
	{

        [XmlAttribute]
        public string id { get; set; }

        [XmlAttribute]
        public string type { get; set; }

        [XmlAttribute("target")]
        public targetColumn target { get; set; }

        [XmlAttribute("source")]
        public sourceColumn[] sources { get; set; }


		public virtual bool isDataFlow
		{
			get
			{
				return "dataflow".Equals(type);
			}
		}

	}

}